hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -227,6 +247,9 @@ protected OmMetadataManagerImpl() {
 
   @Override
   public Table<String, OmKeyInfo> getOpenKeyTable() {
+    if (enableFSPaths && OzoneManagerRatisUtils.isOmLayoutVersionV1()) {
```

Comment:
The idea here is, will use the new tables only if enableFSPaths is true and with V1 version.

Reply:
Good catch @bharatviswa504. Please feel free to add if anything else needed. Thanks again!

Based on our offline discussions, below is the expected behavior for diff requests:

**V1 feature version** : Following ops shouldn't depend on enableFSPaths flag
1) FileCreate  -----> Look into dirTable for parents. Then create entries in openFileTable and on close add it to fileTable.
2) DirCreate  -----> Create entries in dirTable
3) File/DirDelete -> Look into fileTable and dirTable for the keys.
4) File/DirRename-> Look into fileTable and dirTable for the keys.

**V1 feature version & enableFSPaths=true**
1) KeyCreate ---> Look into dirTable for parents. Create entries in openFileTable and on close add it to fileTable.
2) KeyDelete ---> Look into fileTable and dirTable for the keys.
3) KeyRename -> supported only in ozone shell. It should look into fileTable and dirTable for the keys.

**V1 feature version & enableFSPaths=false**
1) KeyCreate ---> Create entries in openKeyTable and on close add it to keyTable.
2) KeyDelete ---> Look into keyTable for the keys.
3) KeyRename -> supported only in ozone shell. It should look into keyTable for the keys.

In this PR, will handle only `FileCreate` request and not provided checks for enableFSPaths in KeyCommit. Will do this changes in latest commit.

Later, I will raise subsequent jiras for handling KeyCreate/KeyCommit and other ops.

Reply:
@rakeshadr , the V1 feature is a new key format and is not compatible with old format. enableFSPaths flag behaves as a switch here, why we remove this? Not so fully understand this.
Without enableFSPaths check, we will directly use new key format by default once V1 feature is supported.
Please correct me if I am wrong.

Reply:
Thanks @linyiqun for the comment. It seems the above comment was not clear. I had done few corrections to the above comment. 

Yes, V1 represents new key format. 

Adding background about  `ozone.om.enable.filesystem.paths` -> this is the config to enable/disable enableFSPaths feature. Basically here the idea is to provide s3/fs inter-op. Please refer jira https://issues.apache.org/jira/browse/HDDS-4097 for more details. If the flag is enabled, then the user given key will be normalized and stored in FS semantics format by OM and it will be 100% FS semantics. If it is false, the key won't be normalized and it will be 100% S3 semantics.

For example, user created a key "/dir1/dir2/dir3/file1" from S3 API. Now,  if the flag is enabled the key will be normalized and create intermediate directories for the file1.

_**More Details:-**_
The cases I mentioned above - **V1 feature version & enableFSPaths=true** is 100% FS semantics and **V1 feature version & enableFSPaths=false** is 100% S3 semantics

Assume the key is /dir1/dir2/dir3/file-1. Again assume V1 feature version enabled and bucketId is 512.
Now, 
**enableFSPaths=true**, which is 100% FS semantics.
It stores as "512/dir1:1025", "1025/dir2:1026" and "1026/dir3:1027" into dirTable and "1027:file1" into openFiletable and on close move it to fileTable

**enableFSPaths=false**, which is 100% S3 semantics.
It stores as "512/dir1/dir2/dir3/file1:1025" into openFileTable and on close move it to fileTable. Here still maintains the parentID/Key format, but the key will be the fullpath and not a normalized path. Here the key can be anything like `/dir1////dir2////dir3///file1`.

Please let me know if any more details needed.

Reply:
Just to add one more point here
this is FS API, so the usage of the flag is of no effect, as this is FS API (File create) not an Object Store API.

But KeyCommitRequest is common for both FS/Object Store we need special handling over there.

Reply:
Sure, I will add special handling in KeyCommit code while implementing KeyCreate request. Hope that make sense to you.

Reply:
Fine with me.

Reply:
Get it, Thanks @rakeshadr and @bharatviswa504 for the very clear explanation!